cmake/SwitchTools.cmake

```diff
@@ -286,9 +286,7 @@ function(add_nro_target target)
 
     # Build the NRO file.
     if(NOT NO_NACP)
-        if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/${target_we}.nacp)
-            __generate_nacp(${target})
-        endif()
+        __generate_nacp(${target})
```

Comment:
A more desirable approach to fixing this would be directly addressing the `CMP0058` issue by specifying the `BYPRODUCTS` option to `add_custom_command` in `__generate_nacp`. Might be reasonable to do that for `__generate_npdm` too.

See [this](https://cmake.org/cmake/help/v3.17/policy/CMP0058.html) article for a more detailed overview of the policy and [this one](https://cmake.org/cmake/help/v3.17/command/add_custom_command.html#command:add_custom_command) for further explanation on `BYPRODUCTS`.

Reply:
That's not the problem. The problem is that in this scenario:

```
cmake -GNinja ..
ninja
cmake -GNinja ..
```

In the second `cmake`, `${CMAKE_CURRENT_BINARY_DIR}/${target_we}.nacp` already exists, leading to the target not being created. However, the .nro still depends on it. Always calling `__generate_nacp` solves this, and Ninja won't regenerate it unless necessary anyway.